src/decode.rs

```diff
@@ -3806,1159 +3806,204 @@ unsafe fn decode_b(
-                    .offset((bx4 + 8) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_116;
-                (*(&mut *((*t.a).skip_mode)
-                    .as_mut_ptr()
-                    .offset((bx4 + 16) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_116;
-                (*(&mut *((*t.a).skip_mode)
-                    .as_mut_ptr()
-                    .offset((bx4 + 24) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_116;
-                let const_val_117: uint64_t = 0 as libc::c_int as uint64_t;
-                (*(&mut *((*t.a).intra).as_mut_ptr().offset((bx4 + 0) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_117;
-                (*(&mut *((*t.a).intra).as_mut_ptr().offset((bx4 + 8) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_117;
-                (*(&mut *((*t.a).intra).as_mut_ptr().offset((bx4 + 16) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_117;
-                (*(&mut *((*t.a).intra).as_mut_ptr().offset((bx4 + 24) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_117;
-                let const_val_118: uint64_t = (0x101010101010101 as libc::c_ulonglong)
-                    .wrapping_mul(b.skip as libc::c_ulonglong)
-                    as uint64_t;
-                (*(&mut *((*t.a).skip).as_mut_ptr().offset((bx4 + 0) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_118;
-                (*(&mut *((*t.a).skip).as_mut_ptr().offset((bx4 + 8) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_118;
-                (*(&mut *((*t.a).skip).as_mut_ptr().offset((bx4 + 16) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_118;
-                (*(&mut *((*t.a).skip).as_mut_ptr().offset((bx4 + 24) as isize) as *mut uint8_t
-                    as *mut alias64))
-                    .u64_0 = const_val_118;
-            }
-            _ => {}
-        }
+
+        let mut set_ctx = |dir: &mut BlockContext, diridx, off, mul, rep_macro: SetCtxFn| {
+            rep_macro(
+                dir.tx_intra.as_mut_ptr() as *mut u8,
+                off,
+                mul * b_dim[2 + diridx] as u64,
+            );
+            rep_macro(dir.mode.as_mut_ptr(), off, mul * DC_PRED as u64);
+            rep_macro(dir.pal_sz.as_mut_ptr(), off, 0);
+
+            // see aomedia bug 2183 for why this is outside `if has_chroma {}`
+            let pal_sz_uv: &mut [u8; 32] = &mut t.pal_sz_uv[diridx];
+            rep_macro(pal_sz_uv.as_mut_ptr(), off, 0);
+
```

Comment:
```suggestion
        let mut set_ctx = |dir: &mut BlockContext, diridx: usize, off, mul, rep_macro: SetCtxFn| {
            rep_macro(
                dir.tx_intra.as_mut_ptr() as *mut u8,
                off,
                mul * b_dim[2 + diridx] as u64,
            );
            rep_macro(dir.mode.as_mut_ptr(), off, mul * DC_PRED as u64);
            rep_macro(dir.pal_sz.as_mut_ptr(), off, 0);
            // see aomedia bug 2183 for why this is outside `if has_chroma {}`
            rep_macro(t.pal_sz_uv[diridx].as_mut_ptr(), off, 0);
```

The type wasn't inferred correctly for `pal_sz_uv` because `diridx` was already inferred.  Specifying that, `pal_sz_uv` is inferred correctly.